pkgs/games/fallout-ce/build.nix

```diff
@@ -0,0 +1,77 @@
+{ cmake
+, fpattern
+, lib
+, SDL2
+, stdenv
+, writeShellScript
+
+, extraBuildInputs ? [ ]
+, extraMeta
+, pname
+, version
+, src
+}:
+
+let
+  launcher = writeShellScript "${pname}" ''
+    set -eu
+    assetDir="''${XDG_DATA_HOME:-$HOME/.local/share}/${pname}"
+    [ -d "$assetDir" ] || mkdir -p "$assetDir"
+    cd "$assetDir"
+
+    notice=0 fault=0
+    requiredFiles=(master.dat critter.dat)
+    for f in "''${requiredFiles[@]}"; do
+      if [ ! -f "$f" ]; then
+        echo "Required file $f not found in $PWD, note the files are case-sensitive"
+        notice=1 fault=1
+      fi
+    done
+
+    if [ ! -d "data/sound/music" ]; then
+      echo "data/sound/music directory not found in $PWD. This may prevent in-game music from functioning."
+      notice=1
+    fi
+
+    if [ $notice -ne 0 ]; then
+      echo "Please reference the installation instructions at https://github.com/alexbatalov/fallout2-ce"
+    fi
+
+    if [ $fault -ne 0 ]; then
+      exit $fault;
+    fi
+
+    exec @out@/bin/${pname}-unwrapped "$@"
+  '';
+in
+stdenv.mkDerivation {
+  inherit pname version src;
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ SDL2 ] ++ extraBuildInputs;
+  hardeningDisable = [ "format" ];
+  cmakeBuildType = "RelWithDebInfo";
+
+  postPatch = ''
+    substituteInPlace third_party/fpattern/CMakeLists.txt \
+      --replace "FetchContent_Populate" "#FetchContent_Populate" \
+      --replace "{fpattern_SOURCE_DIR}" "${fpattern}/include" \
+      --replace "$/nix/" "/nix/"
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D ${pname} $out/bin/${pname}-unwrapped
```

Comment:
A minor thing - we don't want the unwrapped binary (assuming that it doesn't work unless wrapped) to "pollute" the global PATH, so better stick it in $out/libexec and adjust the wrapper.

Reply:
Lol this is exactly what was done before, and I insisted on getting the unwrapped version. The unwrapped binary still works, it's just that it will try to load the assets in the current directory only, while the wrapper will use $XDG_DATA_HOME. I wanted to keep that unwrapped binary in case someone doesn't want assets in $XDG_DATA_HOME and instead wants to make his own wrapper (or no wrapper at all).

Reply:
I don't mind either way, there are a few examples of the `libexec` approach already as this assets dir problem is common for games.

1. [rott](https://github.com/NixOS/nixpkgs/blob/master/pkgs/games/rott/default.nix#L52)
2. [vessel](https://github.com/NixOS/nixpkgs/blob/master/pkgs/games/vessel/default.nix#L64)
3. [openmw](https://github.com/NixOS/nixpkgs/blob/master/pkgs/games/openmw/tes3mp.nix#L139)
4. [nethack](https://github.com/NixOS/nixpkgs/blob/master/pkgs/games/nethack/default.nix#L143)

@TheBrainScrambler Unless you personally have a use case for the `-unwrapped` being on PATH then I'm leaning more towards YAGNI.

Reply:
I don't have an use case myself. It's interesting, I thought there would be more packages exposing their unwrapped version, it's good to know that I was wrong. There's just one lol:
- [0ad](https://github.com/NixOS/nixpkgs/blob/master/pkgs/games/0ad/default.nix#L6)

There are also packages that somehow provide an unwrapped version but which won't run because they are missing libraries otherwise ?
- [prismlauncher](https://github.com/NixOS/nixpkgs/blob/6e1fe01570b8eed68e5969ee6c5fc77159891ed0/pkgs/top-level/all-packages.nix#L38014)
- [xonotic](https://github.com/NixOS/nixpkgs/blob/6e1fe01570b8eed68e5969ee6c5fc77159891ed0/pkgs/top-level/all-packages.nix#L38425)

So perhaps the right approach would be instead to not expose the unwrapped version, and fix this in those packages that I listed too.

Reply:
@peterhoeg moved to libexec